fdbserver/workloads/DataDistributionMetrics.actor.cpp

```diff
@@ -26,52 +26,119 @@
+		ASSERT(nodeCount > 1);
 	}
 
 	static Value getRandomValue() {
 		return Standalone<StringRef>(format("Value/%08d", deterministicRandom()->randomInt(0, 10e6)));
 	}
 
-	ACTOR static Future<Void> _start(Database cx, DataDistributionMetricsWorkload* self) {
-		state int tNum;
-		for (tNum = 0; tNum < self->numTransactions; ++tNum) {
-			loop {
-				state ReadYourWritesTransaction tr(cx);
-				try {
-					state int i;
-					for (i = 0; i < self->writesPerTransaction; ++i) {
-						tr.set(StringRef(format("Key/%08d", tNum * self->writesPerTransaction + i)), getRandomValue());
+	Key keyForIndex(int n) { return doubleToTestKey((double)n / nodeCount, keyPrefix); }
+
+	ACTOR static Future<Void> ddRWClient(Database cx, DataDistributionMetricsWorkload* self) {
+		loop {
+			state ReadYourWritesTransaction tr(cx);
+			state int i;
+			try {
+				for (i = 0; i < self->readPerTx; ++i)
+					wait(success(
+					    tr.get(self->keyForIndex(deterministicRandom()->randomInt(0, self->nodeCount))))); // read
+				for (i = 0; i < self->writePerTx; ++i)
+					tr.set(self->keyForIndex(deterministicRandom()->randomInt(0, self->nodeCount)),
+					       getRandomValue()); // write
+				wait(tr.commit());
+				++self->commits;
+			} catch (Error& e) {
+				wait(tr.onError(e));
+			}
+			tr.reset();
+		}
+	}
+
+	ACTOR Future<Void> resultConsistencyCheckClient(Database cx, DataDistributionMetricsWorkload* self) {
+		state Reference<ReadYourWritesTransaction> tr =
+		    Reference<ReadYourWritesTransaction>(new ReadYourWritesTransaction(cx));
+		loop {
+			try {
+				int startIndex = deterministicRandom()->randomInt(0, self->nodeCount - 1);
+				int endIndex = deterministicRandom()->randomInt(startIndex + 1, self->nodeCount);
+				state Key startKey = self->keyForIndex(startIndex);
+				state Key endKey = self->keyForIndex(endIndex);
+				// Find the last key <= startKey and use as the begin of the range. Since "Key()" is always the starting point, this key selector will never do cross_module_range_read.
+				// In addition, the first key in the result will be the last one <= startKey (Condition #1)
+				state KeySelector begin =
+				    KeySelectorRef(startKey.withPrefix(ddStatsRange.begin, startKey.arena()), true, 0);
+				// Find the last key less than endKey, move forward 2 keys, and use this key as the (exclusive) end of
+				// the range. If we didn't read through the end of the range, then the second last key
+				// in the result will be the last key less than endKey. (Condition #2)
+				state KeySelector end = KeySelectorRef(endKey.withPrefix(ddStatsRange.begin, endKey.arena()), false, 2);
+				Standalone<RangeResultRef> result =
+				    wait(tr->getRange(begin, end, GetRangeLimits(CLIENT_KNOBS->SHARD_COUNT_LIMIT)));
+				// Condition #1 and #2 can be broken if multiple rpc calls happened in one getRange
+				if (result.size() > 1) {
+					if (result[0].key <= begin.getKey() && result[1].key > begin.getKey()) {
```

Comment:
I feel that the debug TraceEvent will be overwhelming. Can we refactor this a bit by only outing the SevError and easily comment out the SevDebug?

For example, change the if() condition to the SevError condition and move the SevDebug message at the end of the 
`if (result.size() > 1) {}`?

Also do we have to "busy" loop{} here? If it adds some delay for each loop iteration, will it still work? (I think so.)

Reply:
Sure. Will comment them out. 
Adds delay should be fine. Is there any issue with busy calling the RPC?

Reply:
It should be fine in simulation. But if we run this in circus (perf test in real cluster), this basically becomes a client keeps issuing txns, which may hurt performance.

Reply:
Yeah, right. Never think about that case. I can parameterize this.